language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -347,20 +347,21 @@ genDefDataType curPkgId conName mod tpls def =
                      ["daml.STATIC_IMPLEMENTS_SERIALIZABLE_CHECK<" <> conName <> ">(" <> conName <> ")"]
             in ((typeDesc, serDesc), Set.unions $ map (Set.setOf typeModuleRef . snd) bs)
         DataEnum enumCons ->
-          let
-            typeDesc =
-                [ "export enum " <> conName <> " {"] ++
-                [ "  " <> cons <> " = " <> "\'" <> cons <> "\'" <> ","
-                | VariantConName cons <- enumCons] ++
-                [ "}"
-                , "daml.STATIC_IMPLEMENTS_SERIALIZABLE_CHECK<" <> conName <> ">(" <> conName <> ")"
-                ]
-            serDesc =
-                ["() => jtv.oneOf<" <> conName <> ">" <> "("] ++
-                ["  jtv.constant(" <> conName <> "." <> cons <> ")," | VariantConName cons <- enumCons] ++
-                [");"]
+          let cs = map unVariantConName enumCons
+              typeDesc = "" : ["  | '" <> cons <> "'" | cons <- cs]
+              -- The complete definition of the companion object.
+              serDesc =
+                [ "export const " <> conName <> ": daml.Serializable<" <> conName <> "> & {"] <>
+                [ "  readonly " <> cons <> ": " <> conName <> ";" | cons <- cs ] ++
+                ["} = {"] ++
+                ["  " <> cons <> ": '" <> cons <> "'," | cons <- cs] ++
+                ["  decoder: () => jtv.oneOf<" <> conName <> ">" <> "("] ++
+                ["      jtv.constant(" <> conName <> "." <> cons <> ")," | cons <- cs] ++
+                ["  ),"] ++
+                ["} as const;"] ++
+                ["daml.STATIC_IMPLEMENTS_SERIALIZABLE_CHECK<" <> conName <> ">(" <> conName <> ")"]
```

Comment:
We already state that `Color` has type `daml.Serializable<Color>` (intersected with more stuff) above. Thus, we can drop this check. If that's the last occurrence of this check (hopfully!), then we can remove the whole `STATIC_IMPLEMENTS_SERIALIZABLE_CHECK` from our libraries.

Reply:
https://github.com/digital-asset/daml/pull/5044. Unfortunately, not the last occurence.

Reply:
although the last occurrence is silly in the same way that this one became. will remove it in this PR freeing the way to delete it from @daml/types.

Reply:
Done in https://github.com/digital-asset/daml/pull/5044 including removal of the definition of the check from `daml-types`.